controller/comments.go

```diff
@@ -70,22 +70,43 @@ func (c *CommentsController) Update(ctx *app.UpdateCommentsContext) error {
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
 	}
-
-	return application.Transactional(c.db, func(appl application.Application) error {
-		cm, err := appl.Comments().Load(ctx.Context, ctx.CommentID)
+	var cm *comment.Comment
+	var wi *workitem.WorkItem
+	// Following transaction verifies if a user is allowed to update or not
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		cm, err = appl.Comments().Load(ctx.Context, ctx.CommentID)
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-
-		if *identityID != cm.CreatedBy {
-			// need to use the goa.NewErrorClass() func as there is no native support for 403 in goa
-			// and it is not planned to be supported yet: https://github.com/goadesign/goa/pull/1030
-			return jsonapi.JSONErrorResponse(ctx, goa.NewErrorClass("forbidden", 403)("User is not the comment author"))
+		wi, err = appl.WorkItems().LoadByID(ctx.Context, cm.ParentID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
 		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	// User is allowed to update if user is creator of the comment OR user is a space collaborator
+	if *identityID == cm.CreatedBy {
```

Comment:
yes, @alexeykazakov got it... please check as done in https://github.com/almighty/almighty-core/pull/1407/commits/a68517e596b79b8ca85435ab373a0a52012878c7
Thanks.